### DISCUSSION TITLE
KeyboardLanguage for Entry and Editor

### DISCUSSION THREAD

**Discussion by jrahma** [xamarin/Xamarin.Forms#3733] [General]
Hi,

Is it possible that you’ll add a KeyboardLanguage property for the Editor and Entry to make a default keyboard starts in English or Arabic or whatever installed language?

User should be able to change the language from they keyboard anytime but this feature is the default language

Thanks
Jassim

**Comment by davidortinau** [comment]
@jrahma I don't believe this is possible on iOS, and I suspect similar for other platforms. I'm reading the [UITextField documentation](https://developer.apple.com/documentation/uikit/uitextfield) and it says 

```
The default language of the device affects the keyboard that pops up with the text field
```

There is no property to declare a language preference. So by default the keyboard will be whatever the user prefers.

From the keyboard details I also don't see a way to tell it the preferred language. There are the UIKeyboardType options: https://docs.microsoft.com/en-us/dotnet/api/uikit.uikeyboardtype?view=xamarin-ios-sdk-12
